x-pack/elastic-agent/pkg/fleetapi/acker/lazy/lazy_acker.go

```diff
@@ -36,19 +38,34 @@ func NewAcker(baseAcker batchAcker, log *logger.Logger) *Acker {
 }
 
 // Ack acknowledges action.
-func (f *Acker) Ack(ctx context.Context, action fleetapi.Action) error {
+func (f *Acker) Ack(ctx context.Context, action fleetapi.Action) (err error) {
+	span, ctx := apm.StartSpan(ctx, "ack", "app.internal")
+	defer func() {
+		if err != nil {
+			apm.CaptureError(ctx, err).Send()
+		}
```

Comment:
[Suggestion]
It's safe to call `apm.CaptureError` with a nil error and then call `Send`. Therefore it can be simplified.
```suggestion
		apm.CaptureError(ctx, err).Send()
```